### ISSUE TITLE
Record Archives error on 2.10.4

### ISSUE THREAD

**Issue by franciscocosta** [web2py/web2py#946]
**Error snapshot**
<type 'exceptions.TypeError'>(isinstance() arg 2 must be a class, type, or tuple of classes and types)

**Traceback**
Traceback (most recent call last):
  File "/home/franciscocosta/www/web2py/gluon/restricted.py", line 227, in restricted
    exec ccode in environment
  File "/home/franciscocosta/www/web2py/applications/emprega/views/historico.html", line 438, in <module>
  File "/home/franciscocosta/www/web2py/gluon/sqlhtml.py", line 3253, in __init__
    r = represent(field, r or [], record)
  File "/home/franciscocosta/www/web2py/gluon/sqlhtml.py", line 68, in represent
    return f(value)
  File "/home/franciscocosta/www/web2py/gluon/packages/dal/pydal/helpers/methods.py", line 268, in __call__
    if isinstance(db._adapter, GoogleDatastoreAdapter):
TypeError: isinstance() arg 2 must be a class, type, or tuple of classes and types

**Code listing**
```
        if not value:
            return None
        from ..adapters import GoogleDatastoreAdapter
        refs = None
        db, id = self.ref._db, self.ref._id
        if isinstance(db._adapter, GoogleDatastoreAdapter):

            def count(values):
                return db(id.belongs(values)).select(id)
            rx = range(0, len(value), 30)
            refs = reduce(lambda a, b: a & b, [count(value[i:i+30])
```

This was working fine on 2.10.3
My database is MongoDB

In the controller:
```
archives = db(db.jobs_archive.current_record == job.id).select(orderby=~db.jobs_archive.modified_on)
```

In the view:
```
{{=SQLTABLE(archives, linkto, upload, orderby=True, _class='table table-bordered table-striped sortable')}}
```

**Comment by ilvalle** [comment]
This is due to GoogleDatastoreAdapter being None.
Can you try to replace the failing line with ```if db._adapter.dbengine == 'google:datastore'```

**Comment by skoa-h** [comment]
Web2py 2.12.2-stable+timestamp.2015.08.09.14.29.44 is still broken because it apparently uses pyDal v.15.03 and the fix is tagged for v15.07. When is web2py planned to be updated?
